.golangci.yml

```diff
@@ -31,6 +31,9 @@ linters-settings:
   errorlint:
     # these are still common in Go: for instance, exit errors.
     asserts: false
+    # Forcing %w in error wrapping forces authors to make errors part of their package APIs.
+    # Also see Hyrums Law.
+    errorf: false
```

Comment:
This sounds like the default. Any reason to set it explicitly?
https://github.com/polyfloyd/go-errorlint/blob/master/README.md#examples

Reply:
The root project does, but `golangci-lint` has it on by default. See https://golangci-lint.run/usage/linters/#errorlint. I had Github action errors as a result.

Reply:
This file is managed centrally via github.com/tinkerbell/lint-install in an effort to have all the go code passing the same linter settings and thus hopefully look/feel similar. If we're to disable this rule we should do that there. I'm not sure I agree about disabling this btw, if you do send this change over there please explain further how this ends up displaying Hyrums Law.

Reply:
Errors are part of the package API. That means if you expose a Kubernetes specific error the Kubernetes specific error is part of the packages API whether you want it or not. That should really be a conscious decision by the package author, not mandated by a linter.

Hyrum's law comes into play because consumers become dependent on that error type and that was never intended making things harder to change in the future.

It's less relevant for internal packages but I typically advocate for the rationale above anyway.

Reply:
Fwiw errorlint even calls this out as a trade off.

Reply:
Removed and raised in the `lint-install` repo.